src/sb.cpp

```diff
@@ -57,4 +58,27 @@ int Sb::CastOffEncoding(FunctorParams *functorParams)
     return FUNCTOR_SIBLINGS;
 }
 
+int Sb::CastOffSystems(FunctorParams *functorParams)
+{
+    CastOffSystemsParams *params = vrv_params_cast<CastOffSystemsParams *>(functorParams);
+    assert(params);
+    if (params->m_smart) {
+        // Get the last measure of the currentSystem
+        Measure *measure
+            = dynamic_cast<Measure *>(params->m_currentSystem->GetChild(params->m_currentSystem->GetChildCount() - 1));
+        if (measure != NULL) {
+            int measureRightX = measure->GetDrawingX() + measure->GetWidth() - params->m_shift;
+            // LogDebug("ratio: %f\n", (float)measureRightX / (float)params->m_systemWidth);
+            if (measureRightX > params->m_systemWidth * .66) {
```

Comment:
Instead of hard coding this we could use the value from  `--min-last-justification`, as it basically does the same for the last system as if you'd put another `sb` end the end of the MEI file.

Reply:
Thanks for bringing this up! I'm not sure `--min-last-justification` would be the right thing to change. In my application, I'm always setting `--min-last-justification` to 0, so that the last system is always justified to fill the full width. But I do see the logic in the argument.

But thinking about that value (`--encoded-sb-thresh`? and the [corollary](https://github.com/rism-ch/verovio/issues/1931#issuecomment-759147574) `--encoded-pb-thresh`) and whether it should be configurable, I think it starts to replace the following breaks modes:

* `--breaks=encoded` almost equivalent to `--breaks=smart --encoded-sb-thresh=0 --encoded-pb-thresh=0` (always use encoded breaks (notably different, as `--breaks=encoded` **will** overcompress systems like https://user-images.githubusercontent.com/12452738/103819157-dd3b3380-5037-11eb-8b82-0e8396517b68.png).
* `--breaks=line` almost equivalent to `--breaks=smart --encoded-sb-thresh=1 --encoded-pb-thresh=0` (same difference as above, with the overcompressing)
* `--breaks=smart`: `--breaks=smart --encoded-sb-thresh=0.66 --encoded-pb-thresh=0.66` (considering changing the 'smart' hard-coded value to 0.51; which is why I'd prefer this to be a variable)
* `--breaks=auto`: `--breaks=smart --encoded-sb-thresh=1 --encoded-pb-thresh=1` (only use encoded breaks when the page is filled up, i.e. never use encoded breaks).

So, then I think there are a couple of options (I guess I'll list from least desirable to most desirable)
1. Remove ability to overcompress music with encoded line breaks, remove/deprecate `--breaks` flag.
2. Use `--min-last-justification` for this value (possibly rename? to `--min-width-justification`?). Add `--min-height-justification` for the "smart" `<pb>` case. (maybe replace `--justify-vertically` flag? fix vertical justification issues?)
3. remove `--breaks=smart` mode, have `--encoded-sb-thresh` and `--encoded-pb-thresh` default to 1 for auto mode
4. Add `--smart-encoded-sb-thresh` (and `--smart-encoded-pb-thresh`) (defaulting to .6 or .5-ish): Flags which are only effective when you have `--breaks=smart`

Reply:
I would prefer to go for option 4. I also find it useful to have options grouped alphabetically as much as possible, so I would suggest `--breaks-smart-sb` for the threshold value.

At this stage I am not absolutely clear why you would need a pb threshold since verovio does not compress pages vertically when there is not enough space in vertical justification. It does only expand. It means that anything below 1.0 would have no effect as it stands. So I would suggest to finalize the implementation with sb first.